admin-dev/themes/new-theme/js/pages/order/view/order-product-renderer.js

```diff
@@ -0,0 +1,192 @@
+  }
+
+  resetAddRow() {
+    $(OrderViewPageMap.productAddIdInput).val('');
+    $(OrderViewPageMap.productSearchInput).val('');
+    $(OrderViewPageMap.productAddCombinationsSelect).val('');
+    $(OrderViewPageMap.productAddPriceTaxExclInput).val('');
+    $(OrderViewPageMap.productAddPriceTaxInclInput).val('');
+    $(OrderViewPageMap.productAddQuantityInput).val('');
+    $(OrderViewPageMap.productAddAvailableText).html('');
+    $(OrderViewPageMap.productAddLocationText).html('');
+    $(OrderViewPageMap.productAddActionBtn).attr('disabled', 'disabled');
+  }
+
+  resetEditRow(orderProductId) {
+    const $productRow = $(OrderViewPageMap.productsTableRow(orderProductId));
+    const $productEditRow = $(OrderViewPageMap.productsTableRowEdited(orderProductId));
+    $productEditRow.remove();
+    $productRow.removeClass('d-none');
+  }
+
+  paginate(orderId, numPage, results) {
+    this.paginateUpdateControls(numPage);
+    // Remove all rows...
+    $(OrderViewPageMap.productsTable).find('tr[id^="orderProduct_"]').remove();
+    // ... and recreate them
+    results.products.forEach(result => this.paginateRowCreate(result, orderId));
+  }
+
+  paginateUpdateControls(numPage) {
+    const totalPage = $(OrderViewPageMap.productsTablePagination).find('li.page-item').length - 2;
+    $(OrderViewPageMap.productsTablePagination).find('.active').removeClass('active');
+    $(OrderViewPageMap.productsTablePagination).find(`li:has(> [data-page="${numPage}"])`).addClass('active');
+    $(OrderViewPageMap.productsTablePaginationPrev).removeClass('disabled');
+    if (numPage === 1) {
+      $(OrderViewPageMap.productsTablePaginationPrev).addClass('disabled');
+    }
+    $(OrderViewPageMap.productsTablePaginationNext).removeClass('disabled');
+    if (numPage === totalPage) {
+      $(OrderViewPageMap.productsTablePaginationNext).addClass('disabled');
+    }
+  }
+
+  paginationAddPage(numPage) {
+    const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+    $tablePagination.data('numPages', numPage);
+    const $linkPagination = $(OrderViewPageMap.productsTablePaginationTemplate).clone();
+    $linkPagination.find('span').attr('data-page', numPage);
+    $linkPagination.find('span').html(numPage);
+    $(OrderViewPageMap.productsTablePaginationTemplate).before($linkPagination.removeClass('d-none'));
+  }
+
+  paginationRemovePage(numPage) {
+    const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+    const numPages = $tablePagination.data('numPages');
+    $tablePagination.data('numPages', numPages - 1);
+    $(OrderViewPageMap.productsTablePagination).find(`li:has(> [data-page="${numPage}"])`).remove();
+  }
+
+  paginateRowCreate(result, orderId) {
+    const $productRow = $(OrderViewPageMap.productTemplateRow).clone();
+    $productRow.attr('id', `orderProduct_${result.orderDetailId}`);
+    if (result.imagePath) {
+      $productRow.find('td.cellProductImg img').attr('src', result.imagePath);
```

Comment:
Don't find the same thing many times `productRow.find('td.cellProductImg img')` it reload the selector everytime for nothing

Reply:
:heavy_check_mark: